app/vmagent/remotewrite/remotewrite.go

```diff
@@ -485,6 +484,16 @@ func tryPush(at *auth.Token, wr *prompbmarshal.WriteRequest, forceDropSamplesOnF
 			rowsCountAfterRelabel := getRowsCount(tssBlock)
 			rowsDroppedByGlobalRelabel.Add(rowsCountBeforeRelabel - rowsCountAfterRelabel)
 		}
+		tssBlockTmp := tssBlock[:0]
+		for _, ts := range tssBlock {
+			if storagelimits.ExceedingLabels(ts.Labels) {
```

Comment:
I think condition should be inverted to `if !storagelimits.ExceedingLabels(ts.Labels)` if labels are not exceeding limits.